DataFormats/PatCandidates/interface/PackedCandidate.h

```diff
@@ -664,13 +652,26 @@ namespace pat {
     reco::VertexRef::key_type pvRefKey_;
 
     /// IP covariance	
-    CMS_THREAD_GUARD(vertex_) mutable float dxydxy_, dzdz_, dxydz_,dlambdadz_,dphidxy_,dptdpt_,detadeta_,dphidphi_;
+    CMS_THREAD_GUARD(vertex_) mutable reco::TrackBase::CovarianceMatrix m_;
     uint8_t packedHits_, packedLayers_; // packedLayers_ -> layers with valid hits; packedHits_ -> extra hits beyond the one-per-layer implied by packedLayers_
+    
     /// track quality information
     uint8_t normalizedChi2_; 
-//    uint8_t numberOfPixelHits_;
-  //  uint8_t numberOfHits_;
-    
+    uint16_t covarianceVersion_;
+    uint16_t covarianceSchema_;
+    CMS_THREAD_SAFE static CovarianceParameterization covarianceParameterization_;
+    //static std::atomic<CovarianceParameterization*> covarianceParameterization_;
+    static std::once_flag covariance_load_flag;
+    const CovarianceParameterization & covarianceParameterization() const {
+	std::call_once(covariance_load_flag,[](int v) { covarianceParameterization_.load(v); } ,covarianceVersion_ );
+        if(covarianceParameterization_.loadedVersion() != covarianceVersion_ )
+        {
+          std::cerr << "Attempting to load multiple covariance version in same process. This is not supported." << std::endl;
```

Comment:
@gpetruc was actually suggesting to be able to handle different version in same job so rather than throwing/aborting on requests for new version we should handle the multiple version loading/caching. So I'd rather not spend time fixing this as the logic is going to change anyhow and the abort is going to be removed